IaC/pmm.cd/JenkinsStack.yml

```diff
@@ -0,0 +1,839 @@
+                  curl https://letsencrypt.org/certs/isrgrootx1.pem                          > /etc/nginx/ssl/ca-certs.pem
+                  curl https://letsencrypt.org/certs/lets-encrypt-x1-cross-signed.pem       >> /etc/nginx/ssl/ca-certs.pem
+                  curl https://letsencrypt.org/certs/letsencryptauthorityx1.pem             >> /etc/nginx/ssl/ca-certs.pem
+                  curl https://www.identrust.com/certificates/trustid/root-download-x3.html >> /etc/nginx/ssl/ca-certs.pem
+              }
+
+              setup_nginx() {
+                  if [ ! -f /mnt/$JENKINS_HOST/ssl/nginx.conf ]; then
+                      cat <<-EOF | tee /mnt/$JENKINS_HOST/ssl/nginx.conf
+              			upstream jenkins {
+              			  server 127.0.0.1:8080 fail_timeout=0;
+              			}
+
+              			server {
+              			  listen 80;
+              			  server_name $JENKINS_HOST;
+
+              			  # letsencrypt certificates validation
+              			  location /.well-known {
+              			    alias /usr/share/nginx/html/.well-known;
+              			  }
+
+              			  # or redirect to https
+              			  if (\$uri !~* ^/.well-known) {
+              			    return 301 https://\$host\$request_uri;
+              			  }
+              			}
+
+              			server {
+              			  listen 443 ssl;
+              			  server_name $JENKINS_HOST;
+
+              			  ssl_certificate             /etc/nginx/ssl/certificate.crt;
+              			  ssl_certificate_key         /etc/nginx/ssl/certificate.key;
+              			  ssl_trusted_certificate     /etc/nginx/ssl/ca-certs.pem;
+              			  ssl_dhparam                 /etc/nginx/ssl/dhparam.pem;
+
+              			  resolver                    8.8.8.8 ipv6=off;
+              			  lua_ssl_trusted_certificate /etc/pki/tls/certs/ca-bundle.crt;
+              			  lua_ssl_verify_depth        3;
+              			  satisfy                     any;
+
+              			  location / {
+              			    proxy_set_header        Host \$host:\$server_port;
+              			    proxy_set_header        X-Real-IP \$remote_addr;
+              			    proxy_set_header        X-Forwarded-For \$proxy_add_x_forwarded_for;
+              			    proxy_set_header        X-Forwarded-Proto \$scheme;
+              			    proxy_redirect http:// https://;
+              			    proxy_pass              http://jenkins;
+              			    # Required for new HTTP-based CLI
+              			    proxy_http_version 1.1;
+              			    proxy_request_buffering off;
+              			    proxy_buffering off; # Required for HTTP-based CLI to work over SSL
+              			    # workaround for https://issues.jenkins-ci.org/browse/JENKINS-45651
+              			    add_header 'X-SSH-Endpoint' '$JENKINS_HOST:50022' always;
+                                    rewrite ^/$ $scheme://$server_name/blue/pipelines/ last;
```

Comment:
Broken indentation :)

Reply:
Thank you, my local IDE creates this, will fix by holy vim